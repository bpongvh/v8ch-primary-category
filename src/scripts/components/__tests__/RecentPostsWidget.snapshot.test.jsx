import React from 'react';
import renderer from 'react-test-renderer';
import RecentPostsWidget from '../RecentPostsWidget';

jest.mock('../RecentPostListItem', () => () => (
  <li className="mock-recent-post-list-item">
    Mock RecentPostListItem
  </li>
));

describe('RecentPostsWidget Snapshot', () => {
  it('renders correctly', () => {
    const posts = [
      {
        link: 'http://some.url.local/1',
        publishedAt: new Date('2018-07-08T12:00:00.000Z').toISOString(),
        title: 'Post Title #1',
      },
      {
        link: 'http://some.url.local/2',
        publishedAt: new Date('2018-07-08T11:00:00.000Z').toISOString(),
        title: 'Post Title #2',
      },
    ];
    const tree = renderer
      .create(<RecentPostsWidget categoryName="Category Name" posts={posts} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
