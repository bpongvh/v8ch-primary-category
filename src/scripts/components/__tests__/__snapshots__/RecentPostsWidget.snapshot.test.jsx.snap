// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentPostsWidget Snapshot renders correctly 1`] = `
Array [
  <h2
    className="widget-title"
  >
    Previously in Category Name:
  </h2>,
  <ul>
    <li
      className="mock-recent-post-list-item"
    >
      Mock RecentPostListItem
    </li>
    <li
      className="mock-recent-post-list-item"
    >
      Mock RecentPostListItem
    </li>
  </ul>,
]
`;
